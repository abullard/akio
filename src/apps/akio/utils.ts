import fs from 'fs';
import path from 'path';
import { glob } from 'glob';

export const getPkgManager = () => {
    if (fs.existsSync('pnpm-lock.yaml')) return 'pnpm';
    if (fs.existsSync('yarn.lock')) return 'yarn';

    return 'npm';
};

const loadJson = async (path: string) => await import(path, { with: { type: 'json' } });

export const readAllPkgJsons = async () => {
    const paths = await glob('**/package.json', {
        signal: AbortSignal.timeout(1000),
        ignore: 'node_modules/**',
        absolute: true
    });

    const allPkgJsonContents = await Promise.all(paths.map(loadJson));
    for (const contents in allPkgJsonContents) {
        
    }

    // original impl
    return;
}
