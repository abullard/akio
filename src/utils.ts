import fs from 'fs';
import path from 'path';

export const getPkgManager = () => {
    if (fs.existsSync('pnpm-lock.yaml')) return 'pnpm';
    if (fs.existsSync('yarn.lock')) return 'yarn';

    return 'npm';
};

export const readPackageJson = () => {
    const pkgPath = path.resolve(process.cwd(), 'package.json');
    return JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
}
