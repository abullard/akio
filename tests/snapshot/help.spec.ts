import { describe, it, expect } from 'vitest';
import { spawnWrapper } from '../utils/test-utils';

describe('search2.spec.ts', () => {
    it('should output help menu', async () => {
        const cmd = 'pnpm';
        const args = ['akio', '--help'];

        const { stdout } = await spawnWrapper(cmd, args);

        expect(stdout).toMatchSnapshot();
    });
});
