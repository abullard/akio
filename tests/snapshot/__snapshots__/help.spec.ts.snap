// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`search2.spec.ts > should output help menu 1`] = `
"
> node ./dist/index.cjs --help


ðŸª„ [33makio[0m

  Surface your [32mpackage.json[0m scripts from the depths of your [35mmonorepo[0m. 

CLI Opts

  -i, --input           Do not prompt for an input comand.                      
  -f, --format          Do not include emoji or color formatting.               
  -d, --descriptions    Do not warn if package.json "scriptDescriptions" are    
                        missing.                                                
  -h, --help            Display this usage guide.                               

  https://npmjs.com/@abullard/akio 
"
`;
